kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -105,6 +105,55 @@ public class TableConfig<T> {
                     "needs to be a long."
             );
 
+    /*
+     * This table property is used to track the commit-coordinator name for this table. If this
+     * property is not set, the table will be considered as file system table and commits will be
+     * done via atomically publishing the commit file.
+     */
+    public static final TableConfig<Optional<String>> COORDINATED_COMMITS_COORDINATOR_NAME =
+            new TableConfig<>(
+                    "delta.coordinatedCommits.commitCoordinator-preview",
+                    null, /* default values */
+                    Optional::ofNullable,
+                    value -> true,
+                    "The commit-coordinator name for this table. This is used to determine " +
+                            "which implementation of commit-coordinator to use when committing " +
+                            "to this table. If this property is not set, the table will be " +
+                            "considered as file system table and commits will be done via " +
+                            "atomically publishing the commit file."
+            );
+
+    /*
+     * This table property is used to track the configuration properties for the commit coordinator
+     * which is needed to build the commit coordinator client.
+     */
+    public static final TableConfig<Map<String, String>> COORDINATED_COMMITS_COORDINATOR_CONF =
+            new TableConfig<>(
+                    "delta.coordinatedCommits.commitCoordinatorConf-preview",
+                    null, /* default values */
+                    v -> {
+                        throw new UnsupportedOperationException("Not implemented yet");
+                    },
+                    value -> true,
+                    "A string-to-string map of configuration properties for the" +
+                            " coordinated commits-coordinator."
+            );
+
+    /*
+     * This table property is used to track the table managed by commit-coordinator internally.
```

Comment:
This property is used by the commit coordinator to uniquely identify and manage the table internally.